spec-tests/src/test/java/org/candlepin/spec/pools/PostBindBonusPoolSpecTest.java

```diff
@@ -0,0 +1,1198 @@
+        String ownerKey = owner.getKey();
+
+        ConsumerDTO cpUser = Consumers.random(owner, ConsumerTypes.Candlepin);
+        cpUser = adminClient.consumers().createConsumer(cpUser);
+        ApiClient candlepinClient = ApiClients.trustedConsumer(cpUser.getUuid());
+
+        ConsumerDTO guest = Consumers.random(owner, ConsumerTypes.System);
+        guest.setUuid(StringUtil.random("guest"));
+        guest.setFacts(Map.of("virt.is_guest", "true", "virt.uuid", guest.getUuid()));
+        guest = adminClient.consumers().createConsumer(guest);
+        ApiClient guestClient = ApiClients.trustedConsumer(guest.getUuid());
+
+        ProductDTO hostLimitedUnlimitedVirtProd =
+            createHostLimitedUnlimitedVirtProduct(adminClient, ownerKey);
+        PoolDTO hostAndVirtLimitedProdMasterPool = Pools.randomUpstream(hostLimitedUnlimitedVirtProd)
+            .quantity(-1L);
+        hostAndVirtLimitedProdMasterPool = adminClient.owners()
+            .createPool(ownerKey, hostAndVirtLimitedProdMasterPool);
+        PoolDTO hostLimitedUnlimitedVirtBonusPool =
+            getBonusPool(adminClient, ownerKey, hostAndVirtLimitedProdMasterPool.getSubscriptionId());
+        hostAndVirtLimitedProdMasterPool = adminClient.pools()
+            .getPool(hostAndVirtLimitedProdMasterPool.getId(), null, null);
+        assertEquals(-1, hostAndVirtLimitedProdMasterPool.getQuantity());
+
+        ProductDTO unlimitedProd = createUnlimitedProduct(adminClient, ownerKey, 4);
+        PoolDTO unlimitedMasterPool = adminClient.owners()
+            .createPool(ownerKey, Pools.randomUpstream(unlimitedProd).quantity(-1L));
+
+        // consume master pool in any quantity
+        candlepinClient.consumers().bindPool(cpUser.getUuid(), unlimitedMasterPool.getId(), 1000);
+        unlimitedMasterPool = adminClient.pools().getPool(unlimitedMasterPool.getId(), null, null);
+        assertEquals(-1, unlimitedMasterPool.getQuantity());
+
+        // unmapped_guest pool quantity is expected to be unlimited
+        hostLimitedUnlimitedVirtBonusPool = adminClient.pools()
+            .getPool(hostLimitedUnlimitedVirtBonusPool.getId(), null, null);
+        assertEquals(-1, hostLimitedUnlimitedVirtBonusPool.getQuantity());
+
+        guestClient.consumers().bindPool(guest.getUuid(), hostLimitedUnlimitedVirtBonusPool.getId(), 500);
+
+        hostLimitedUnlimitedVirtBonusPool = adminClient.pools()
+            .getPool(hostLimitedUnlimitedVirtBonusPool.getId(), null, null);
+        assertEquals(-1, hostLimitedUnlimitedVirtBonusPool.getQuantity());
+    }
+
+    @Nested
+    @OnlyInHosted
+    public class LockedEntityTests {
+        private ApiClient adminClient;
+        private OwnerDTO owner;
+        private String ownerKey;
+        private PoolDTO unlimitedProdMasterPool;
+        private PoolDTO hostAndVirtLimitedProdMasterPool;
+
+        @BeforeEach
+        void setup() throws Exception {
```

Comment:
You can apply the approach you use above to the setup as well. It will make setup steps more obvious. E.g.

```java
        @BeforeEach
        void setup() {
            SubscriptionDTO unlimitedProdSub = createUnlimitedProdSub();
            SubscriptionDTO limitedHostVirtSub = createLimitedHostVirtSub();

            AsyncJobStatusDTO job = adminClient.owners().refreshPools(ownerKey, false);
            AsyncJobStatusDTO status = adminClient.jobs().waitForJob(job.getId());
            assertEquals("FINISHED", status.getState());

            unlimitedProdMasterPool = getPoolBySubId(unlimitedProdSub.getId());
            assertEquals(-1, unlimitedProdMasterPool.getQuantity());

            hostAndVirtLimitedProdMasterPool = getPoolBySubId(limitedHostVirtSub.getId());
            assertEquals(-1, hostAndVirtLimitedProdMasterPool.getQuantity());
        }
```

Reply:
Yeah, I agree! Updated